src/Compilers/CSharp/Portable/Symbols/MethodGroupConversionCaches/MethodGroupConversionCacheFrame.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CodeGen;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// This class holds ALL types of delegates that are coverted from ONE target method
+    /// Type parameters are generated for: <see cref="GetTypeArgumentsFromConversion(TypeSymbol, MethodSymbol)"/>
+    ///     1. All type parameters from the containing types of the target method
+    ///     2. All type parameters from the target method
+    ///     3. The delegate type
+    /// </summary>
+    /// <remarks>
+    /// This symbol is created in lowering phrase, and is collected and translated in emit phrase.
+    /// </remarks>
+    internal sealed class MethodGroupConversionCacheFrame : NamedTypeSymbol
```

Comment:
If the method is not generic one might wonder why the caching field is wrapped into a type at all. Couldn't we just emit the static fields into the containing type? The problem is with type load semantics. The types referred to by the delegates would be loaded when the containing type is loaded, not when the method is first called.